src/peer_digest.cc

```diff
@@ -538,7 +538,14 @@ peerDigestFetchReply(void *data, char *buf, ssize_t size)
 
             assert(fetch->old_entry->mem_obj->request);
 
-            Store::Root().updateOnNotModified(fetch->old_entry, *fetch->entry);
+            try {
+                Store::Root().updateOnNotModified(fetch->old_entry, *fetch->entry);
+            } catch (...) {
+                static const char *reason = "error when updating entry";
+                debugs(20, DBG_IMPORTANT, reason << *fetch->old_entry << ": " << CurrentException);
+                peerDigestFetchAbort(fetch, buf, reason);
+                return -1;
```

Comment:
Similar code below says `/* XXX -1 will abort stuff in ReadReply! */`. Does that XXX mark a Squid bug of some kind? Should we mark it here as well?

Reply:
Understood. Since we cannot tell what that XXX really means and the (already inconsistent with regard to that XXX marking) code looks otherwise OK to us, let's leave this `return -1` as is. I am resolving this change request.